<!-- Launch all nodes of RNE -->

<launch>
	<arg name="update_frequency" default="20" />
	<arg name="sensor_max_range" default="5" />
	<arg name="sensor_min_range" default="0.5" />
	<arg name="sensor_height" default="0.5" />
	<arg name="sensor_size" default="0.1" />
	<arg name="robot_frame" default="base_footprint" />
	<arg name="robot_width" default="1.0" />
	<arg name="robot_radius" default="1.0" />
	<arg name="delta_radius" default="0.1" />
	<arg name="delta_phi" default="10" />
	<arg name="delta_theta" default="10" />
	<arg name="sensor_horizontal_fov" default="360" /> 
	<arg name="sensor_vertical_fov_bottom" default="180" />
	<arg name="sensor_vertical_fov_top" default="0" />
	<arg name="octomap_topic" default="octomap_binary" />
	<arg name="occupancy_grid_topic" default="map" />	
	<arg name="exploration_finished_timer_duration" default="1.0" />
	<arg name="check_init_position" default="false" />
	<arg name="oc_resolution" default="0.1" />
	<arg name="grid_map_resolution" default="0.05" />
	<arg name="local_sampling_radius" default="5" />
	<arg name="min_edge_distance" default="1.0" />
	<arg name="max_edge_distance" default="2.0" />	
	<arg name="nearest_node_tolerance" default="0.1" />		
	<arg name="max_node_height_difference" default="1.0" />	
	<arg name="max_consecutive_failed_goals" default="5.0" />

	<node pkg="rrg_nbv_exploration" type="rneNode" name="rneNode"
		output="screen">
		<param name="update_frequency" type="double" value="$(arg update_frequency)" />
		<param name="sensor_max_range" type="double" value="$(arg sensor_max_range)" />
		<param name="sensor_height" type="double" value="$(arg sensor_height)" />
		<param name="robot_radius" type="double" value="$(arg robot_radius)" />
		<param name="robot_width" type="double" value="$(arg robot_width)" />
		<param name="exploration_finished_timer_duration" type="double" value="$(arg exploration_finished_timer_duration)" />
		<param name="octomap_topic" type="string" value="$(arg octomap_topic)" />
		<param name="occupancy_grid_topic" type="string" value="$(arg occupancy_grid_topic)" />
		<param name="robot_frame" type="string" value="$(arg robot_frame)" />
		<param name="check_init_position" type="bool" value="$(arg check_init_position)" />
		<param name="grid_map_resolution" type="double" value="$(arg grid_map_resolution)" />
		<param name="local_sampling_radius" type="double" value="$(arg local_sampling_radius)" />	
		<param name="min_edge_distance" type="double" value="$(arg min_edge_distance)" />
		<param name="max_edge_distance" type="double" value="$(arg max_edge_distance)" />
		<param name="nearest_node_tolerance" type="double" value="$(arg nearest_node_tolerance)" />
		<param name="max_consecutive_failed_goals" type="double" value="$(arg max_consecutive_failed_goals)" />			
	</node>
	
	<node pkg="rrg_nbv_exploration" type="gainCalcNode" name="gainCalcNode"
		output="screen">
		<param name="sensor_max_range" type="double" value="$(arg sensor_max_range)" />
		<param name="sensor_min_range" type="double" value="$(arg sensor_min_range)" />
		<param name="sensor_height" type="double" value="$(arg sensor_height)" />
		<param name="sensor_size" type="double" value="$(arg sensor_size)" />
		<param name="delta_phi" type="int" value="$(arg delta_phi)" />
		<param name="delta_theta" type="int" value="$(arg delta_theta)" />
		<param name="delta_radius" type="double" value="$(arg delta_radius)" />
		<param name="sensor_horizontal_fov" type="int" value="$(arg sensor_horizontal_fov)" />
		<param name="sensor_vertical_fov_bottom" type="int" value="$(arg sensor_vertical_fov_bottom)" />
		<param name="sensor_vertical_fov_top" type="int" value="$(arg sensor_vertical_fov_top)" />
		<param name="octomap_topic" type="string" value="$(arg octomap_topic)" />
		<param name="oc_resolution" type="double" value="$(arg oc_resolution)" />
		<param name="max_node_height_difference" type="double" value="$(arg max_node_height_difference)" />
	</node>
	
	<node pkg="rrg_nbv_exploration" type="rneVisualizationNode"
		name="rneVisualizationNode" output="screen" />
</launch>
